spring-security-oauth2/src/main/java/org/springframework/security/oauth2/client/token/grant/client/ClientCredentialsAccessTokenProvider.java

```diff
@@ -41,7 +41,14 @@ public OAuth2AccessToken obtainAccessToken(OAuth2ProtectedResourceDetails detail
 			throws UserRedirectRequiredException, AccessDeniedException, OAuth2AccessDeniedException {
 
 		ClientCredentialsResourceDetails resource = (ClientCredentialsResourceDetails) details;
-		return retrieveToken(request, resource, getParametersForTokenRequest(resource), new HttpHeaders());
+		HttpHeaders headers = new HttpHeaders();
+		request.getHeaders().forEach((key, value) -> {
```

Comment:
The `client_credentials` grant does not require an [Authorization Request](https://tools.ietf.org/html/rfc6749#section-4.4.1) so the passed in `request` (`AccessTokenRequest`) is simply an initialized `DefaultAccessTokenRequest` with empty `headers`.

Hope this makes sense @matbrighi ? I'm going to close this PR since this change is not needed.